File: practica1/P1S3/Java/src/practica1dss3/Filtro1.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package practica1dss3;

/**
 *
 * @author USUARIO
 */
public class Filtro1 implements Filtro{
    
    private Comida comida;
    
    public Filtro1(){
        
    }

    @Override
    public void ejecutar() {
        System.out.println("Se ha descontado el 21% del IVA");
        comida.setPrecio(comida.getPrecio() * 0.79);
    }

    @Override
    public void setComida(Comida c) {
        comida = c;
    }
    
}

File: practica1/P1S3/C++/Filtro1.cpp
#pragma once
#include "Filtro.cpp"
#include "Comida.cpp"
#include <iostream>
#include "Carne.cpp"
using namespace std;

class Filtro1 : public Filtro
{
    public:
    Filtro1(){}

    Filtro1(Comida *c)
    {
        comida = c;
    }

    void ejecutar()
    {
        cout << "Se ha descontado el 21% del IVA" << endl;
        comida->setPrecio(comida->getPrecio() * 0.79);
    }

    void setComida(Comida *c)
    {
        comida = c;
    }
    
};
